From 5bc6c11783e7aa1ab0541704abd48e21e4249df1 Mon Sep 17 00:00:00 2001
From: Jayesh Choudhary <j-choudhary@ti.com>
Date: Fri, 7 Feb 2025 11:09:06 +0530
Subject: [PATCH 20/68] PENDING: arm64: dts: ti: k3-j721s2-main: Make the
 scm_conf node a simple-bus

The scm_conf bus need not be syscon. Now that we have pcie*_ctrl child
nodes for 'ti,syscon-pcie-ctrl' property for pcie nodes, change scm_conf
node to simple-bus.
Also remove reg proerty as it is not needed by "simple-bus".
This gets rid of dtbs warnings while adding clock-controller nodes for
audio_external_refclk.

Signed-off-by: Jayesh Choudhary <j-choudhary@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi b/arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi
index d29a8fb3554e..ce257d31325e 100644
--- a/arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi
@@ -38,9 +38,8 @@ l3cache-sram@200000 {
 		};
 	};
 
-	scm_conf: syscon@104000 {
-		compatible = "ti,j721e-system-controller", "syscon", "simple-mfd";
-		reg = <0x00 0x00104000 0x00 0x18000>;
+	scm_conf: bus@104000 {
+		compatible = "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
 		ranges = <0x00 0x00 0x00104000 0x18000>;
-- 
2.47.2

